lxd/containers.go

```diff
@@ -25,6 +28,26 @@ import (
 	"gopkg.in/lxc/go-lxc.v2"
 )
 
+// #cgo LDFLAGS: -lutil
+/*
+#include <unistd.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <grp.h>
+#include <pty.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+#include <stdio.h>
+#include <termios.h>
+
+void own_pty(int fd) {
+	if (ioctl(fd, TIOCSCTTY, (char *)NULL) == -1)
+		printf("Failed TIOCSCTTY: %s\n", strerror(errno));
+}
+*/
+import "C"
+
```

Comment:
Can we move the C part to share/util.go so we concentrate all of our C wrappers in one place?

Then just make a tiny wrapper function in Go which calls C.own_pty, taking a *os.File and passing file.Fd() to the C function.